### PR TITLE
Fix pull docs build running with a schedule and increase cpp doc timeout to 4h

### PR DESCRIPTION
**Pull request by huydhn**
* After https://github.com/pytorch/pytorch/pull/88373, pull workflow can now be triggered with a schedule. This changes the assumption in the doc build workflow when schedule event is used to determine if the docs should be pushed
* I'll create a follow-up issue to see if it's possible to improve the performance of cpp doc build job.  At the moment, it uses a linux.12xlarge runner and still couldn't finish the job after 3h

### CODE REVIEW

**Review by seemethere — APPROVED** [review]

File: .github/workflows/_docs.yml [hunk 1]
```diff
@@ -48,8 +48,9 @@ jobs:
             # to the next available tier of 12xlarge. So much memory just to generate cpp
             # doc
             runner: linux.12xlarge
-            # Nightly cpp docs take about 150m to finish, and the number is stable
-            timeout-minutes: 180
+            # TODO: Nightly cpp docs take longer and longer to finish (more than 3h now)
+            # Let's try to figure out how this can be improved
```
Comment (seemethere) [inline .github/workflows/_docs.yml:52]: +1000

File: .github/workflows/_docs.yml [hunk 2]
```diff
@@ -95,7 +99,10 @@ jobs:
         timeout-minutes: ${{ matrix.timeout-minutes }}
         id: build-docs
         env:
-          WITH_PUSH: ${{ github.event_name == 'schedule' || startsWith(github.event.ref, 'refs/tags/v') }}
```
Comment (ZainRizvi) [inline .github/workflows/_docs.yml:98]: What was the old logic here trying to do? It seems like it's only trying to push a PR when a scheduled (nightly) job is running or when a release branch (with the v tag) is running

The new behavior seeeeems to just push docs on nightlies. Have we lost the push on release branches or does that still happen?

**Review by ZainRizvi — APPROVED** [review]

File: .github/workflows/_docs.yml (hunk 2 above)
Comment (malfet) [reply .github/workflows/_docs.yml:98]: Hmm, why do we want to change that? I.e. this workflow is run periodically there are no harm in pushing the update, is there?

Comment (huydhn) [reply .github/workflows/_docs.yml:98]: I think the decision to push the doc or not should depend only on the `inputs.push` parameter.  This change actually makes the workflow consistent with the logic on https://github.com/pytorch/pytorch/blob/master/.github/workflows/_docs.yml#L85. The necessary credential to push doc will only be available when `inputs.push` is set to true.

So pushing to a release branch or running pull as a periodic job can both publish the doc as long as the upstream workflow set the `inputs.push` param to true accordingly.

**Comment by huydhn** [comment]
@pytochbot merge

**Pull request closed without merging**